src/preferences/dialog/dlgprefautodj.cpp

```diff
@@ -184,3 +247,52 @@ void DlgPrefAutoDJ::slotEnableAutoDJRandomQueue(int a_iValue) {
                 ConfigValue(1));
     }
 }
+
+void DlgPrefAutoDJ::slotSetAutoDJTransitionUnit(int a_iValue) {
+    if (a_iValue == 0) {
+        // Use seconds
+        m_pConfig->set(ConfigKey("[Auto DJ]", "TransitionUnitBuff"),
+                ConfigValue(0));
+    } else {
+        // Use beats
+        m_pConfig->set(ConfigKey("[Auto DJ]", "TransitionUnitBuff"),
+                ConfigValue(1));
+    }
+}
+
+void DlgPrefAutoDJ::slotEnableAutoDJBPMtoUseComboBox(int a_iValue) {
+    if (a_iValue == 0) {
+        // Using seconds, disable the combo box
+        m_pConfig->set(ConfigKey("[Auto DJ]", "EnableAutoDJBPMToUseBuff"),
+                ConfigValue(0));
+        ComboBoxAutoDjBPMToUse->setEnabled(false);
+    } else {
+        m_pConfig->set(ConfigKey("[Auto DJ]", "EnableAutoDJBPMToUseBuff"),
+                ConfigValue(1));
+        ComboBoxAutoDjBPMToUse->setEnabled(true);
+    }
+}
+
+void DlgPrefAutoDJ::slotSetAutoDJBPMtoUse(int a_iValue) {
+    switch (a_iValue) {
+    case 1:
+        // To-deck's BPM
+        m_pConfig->set(ConfigKey("[Auto DJ]", "BPMToUseBuff"),
```

Comment:
Since you are using an int here, you could use the setValue instead.

Reply:
[Done](https://github.com/mixxxdj/mixxx/pull/2746/commits/61fcc3f4bd44a407707731ab972bc0fd005ab0f1). Using setValue instead of set.